components/NavigationDrawerListItem/index.tsx

```diff
@@ -0,0 +1,33 @@
+import ListItemText from '@mui/material/ListItemText'
+import ListItemIcon from '@mui/material/ListItemIcon'
+import ListItemButton from '@mui/material/ListItemButton'
+// props for the list
+interface ListProps {
+  text: string
+  icon?: React.ReactNode
+  collapsable?: boolean
+}
+
+function NavigationDrawerListItem({
+  collapsable = false,
+  icon,
+  text,
+}: ListProps) {
+  return collapsable ? (
+    <div>
+      <ListItemButton>
+        <ListItemIcon>{icon}</ListItemIcon>
+        <ListItemText
+          primary={text}
+          primaryTypographyProps={{ variant: 'body2' }}
```

Comment:
Instead of a ternary expression to render one component or another, a shorter way to code this `<NavigationDrawerListItem>` component would be to just supply `primaryTypographyProps={{ variant: collapsable ? 'body2' : 'body1' }}` to this component. That way, there are no repeated lines of code.